File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float get_positive_float(string prompt);
int quarter = 25;
int dime = 10;
int nickel = 5;
int penny = 1;
int lookChange(int x);
int greedy(int n, int x);
int m = 0;

//Calculates the minumun ammount of coins to give to the client acording to change.
int main(void)
{
    //Formatś it to int.
    int change = round((100 * (get_positive_float("Change owed?\n"))));
    lookChange(change);
    printf("\n Coins: %i\n", m);
}
//Possitive float verification
float get_positive_float(string prompt)
{
    float n;
    do
    {
        n = get_float("%s", prompt);

    }
    while (n < 0);
    return n;
}
//Checks the change and chooses to what operation to make.
int lookChange(int x)
{
    int n;
    while (x > 0)
    {
        if (x >= quarter)
        {
            x = greedy(1, x);

        }
        else if (x >= dime)
        {
            x = greedy(2, x);

        }
        else if (x >= nickel)
        {
            x = greedy(3, x);

        }
        else if (x >= penny)
        {
            x = greedy(4, x);

        }
        else
        {
            return x;
        }
    }
    return x;
}
//Makes operation acording to the coin using a switch idk
int greedy(int n, int x)
{
    int i = 0;
    switch (n)
    {
        case 1:
            i = (x / quarter);
            m = (m + i);
            x = (x - (i * quarter));
            printf("Quarters: %i", i);
            break;

        case 2:
            i = (x / dime);
            m = (m + i);
            x = (x - (i * dime));
            printf("Dime: %i", i);
            break;

        case 3:
            i = (x / nickel);
            m = (m + i);
            x = (x - (i * nickel));
            printf("Nickel: %i", i);
            break;
        case 4:
            i = (x / penny);
            m = (m + i);
            x = (x - (i * penny));
            printf("Penny: %i", i);
            break;
    }
    return x;
    printf("\n");
}

File: pset6/cash/cash.py
from cs50 import get_float

# Constants to use in the program
quarter = 25
dime = 10
nickel = 5
penny = 1
# Variables to use
change = 0
coins = 0

# Makes the operation to get the reminder and the type of coin used


def greedy(x, div):
    i = 0
    global coins
    i = int((x / div))  # Rounding so that it only takes the whole number before the decimals
    x = (x % div)
    coins = (coins + i)
    return x

# Checkes and selects the operation to be made for change to reach 0


def lookChange(x):
    while (x > 0):
        print("Esto es el loop, ", x)
        if (x >= quarter):
            x = greedy(x, quarter)

        elif (x >= dime):
            x = greedy(x, dime)

        elif (x >= nickel):
            x = greedy(x, nickel)

        elif (x >= penny):
            x = greedy(x, penny)
        else:
            return x
    return x


# Validates user input
while(change < 0.1):
    change = round(100*(get_float("Change Owed: ")))
# Calculates minimun number of coins to give
lookChange(change)

# Return coins
print(coins)